aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1223,12 +1222,14 @@ internal class RealAWSCognitoAuthPlugin(
         authStateMachine.getCurrentState { authState ->
             when (val state = authState.authNState) {
                 is AuthenticationState.SignedIn -> {
-                    updateDevice(
-                        (state.deviceMetadata as? DeviceMetadata.Metadata)?.deviceKey,
-                        DeviceRememberedStatusType.Remembered,
-                        onSuccess,
-                        onError
-                    )
+                    GlobalScope.launch {
```

Comment:
runBlocking and GlobalScope both will have the same outcome here since there is nothing to execute after globalScope it will just finish when it finishes while starting a coroutine vs a runblocking will block the thread.